recipes-security/refpolicy/refpolicy-mcs-2.%/policy/modules/system/stubdom-helpers.fc

```diff
@@ -21,3 +21,7 @@
 /usr/lib/xen/bin/atapi_pt_helper        --      gen_context(system_u:object_r:atapi_helper_exec_t,s0)
 /usr/lib/xen/bin/audio_helper(_start)?  --      gen_context(system_u:object_r:audio_helper_exec_t,s0)
 /var/lock/xen-atapi-pt-lock-[0-9]+_[0-9]+_[0-9]+_[0-9]+ -- gen_context(system_u:object_r:atapi_helper_lock_t,s0)
+
+/usr/sbin/varstored            --      gen_context(system_u:object_r:varstored_exec_t,s0)
+/usr/sbin/varstored-watch      --      gen_context(system_u:object_r:varstored_watch_exec_t,s0)
+/usr/share/varstored(/.*)?     --      gen_context(system_u:object_r:varstored_auth_t,s0)
```

Comment:
`--` means it only applies to files, so directory /usr/share/varstored will remain usr_t.  And that is why below you only have `allow varstored_t varstored_auth_t:file read_file_perms;` without `:dir` .  So you probably want to drop `--` and then add `allow varstored_t varstored_auth_t:dir search_dir_perms` (though it might have to be list_dir_perms depending on what varstored_t does).

Reply:
Don't think I ever realized that. Always thought it was a separator for the refpolicy parser.